test/helpers/up-client.js

```diff
@@ -0,0 +1,81 @@
+import { connect } from '@ucanto/client'
+import { CAR, CBOR, HTTP } from '@ucanto/transport'
+import * as DID from '@ipld/dag-ucan/did'
+import * as Signer from '@ucanto/principal/ed25519'
+import { importDAG } from '@ucanto/core/delegation'
+import { CarReader } from '@ipld/car'
+import { AgentData } from '@web3-storage/access/agent'
+
+import { Client } from '@web3-storage/w3up-client'
+
+/**
+ * Get w3up-client configured with staging endpoints and CI Keys.
+ *
+ * @param {string} uploadServiceUrl
+ * @param {object} [options]
+ * @param {boolean} [options.shouldRegister]
+ */
+export async function getClient(uploadServiceUrl, options = {}) {
+  // Load client with specific private key
+  const principal = Signer.parse(process.env.INTEGRATION_TESTS_UCAN_KEY || '')
+  const data = await AgentData.create({ principal })
+
+  const client = new Client(data, {
+    serviceConf: {
+      upload: getUploadServiceConnection(uploadServiceUrl),
+      access: getAccessServiceConnection()
+    },
+  })
+
+  // Add proof that this agent has been delegated capabilities on the space
+  const proof = await parseProof(process.env.INTEGRATION_TESTS_PROOF || '')
+  const space = await client.addSpace(proof)
+  await client.setCurrentSpace(space.did())
+
+  return client
+}
+
+/** @param {string} data Base64 encoded CAR file */
+async function parseProof (data) {
+  const blocks = []
+  const reader = await CarReader.fromBytes(Buffer.from(data, 'base64'))
+  for await (const block of reader.blocks()) {
+    blocks.push(block)
+  }
+  // @ts-expect-error incompatible CID type versions in dependencies
+  return importDAG(blocks)
+}
```

Comment:
It looks like we cannot because https://github.com/web3-storage/w3protocol/blob/main/packages/access-client/src/encoding.js#L81 and with `w3 delegation create <did_from_ucan-key_command_above> | base64
` we don't have a root.

Reply:
really ? the w3 cli doesnt use w3access to create delegations ?